### PR TITLE
Update for latest Sphinx and pydata theme

### PR DESCRIPTION
**Pull request by bryevdv**
This PR updates our docs build to work with 

* Sphinx 5.0 
* pydata-sphinx-theme 0.9

The minimum requirements for these have also been raised in the environment files (we are the only support consumers of the docs build, we don't need to support multiple versions)

Some notes:

* pydata-sphinx-theme has added its own version switcher. I have jettisoned all of our old version switching code. This requires a new `switcher.json` file since the format is different than our custom `versions.json` file. We will have to leave `versions.json` around on the CDN, so that old versions of docs still have a working switcher. But their switchers will no longer update unless we:
  * keep updating `versions.json` alongside `switcher.json`
  * re-deploy all the `custom.js` files to refer to `switcher.json`
 
  The latter is do-able but probably a bit tedious.  The former is also do-able but annoying. 

* We need a new `navbar-logo.html` template because the theme does not seem to support logos from remote URLs. I will open an issue about that. This was true previously as well, but now the change is at least nicely confined to a single template. 

* Out customized `layout.html` is more simplified. AFIACT everything is rendering as expected now, but I woud like more eyes reviewing the output. ]

* Removed our custom "alert" functions. We have never used it, and the pydata theme also apparently supports this already. 

cc @bokeh/dev  

Edit: the basic switcher is pretty ugly:

Screen Shot 2022-06-26 at 11 05 04

It would be be good to add some minimal CSS styling to make it a little prettier. FYI all the new switcher docs are here: https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/configuring.html#add-a-dropdown-to-switch-between-docs-versions

### CODE REVIEW

**Comment by bryevdv** [comment]
The only issue I have noticed so far: individual gallery pages render with a sidebar taking up space:

Screen Shot 2022-06-26 at 11 11 05

It think that is related to some of the `layout.html` that was removed, but I think some of underlying templates changed, so we need to take another look how to suppress the sidebar on the gallery item pages. (This is a nice to have but I actually could probably live with just keeping the sidebar, if it comes to that).

**Comment by bryevdv** [comment]
Ah, the gallery is because after #12114 all examples (including gallery examples) are now under `docs/examples/`. There is no more `docs/gallery` at all. I can fix the sidebar but @tcmetzger is there any way you know that we might redirect `docs/gallery` links so that old links around the internet still function?

**Comment by bryevdv** [comment]
I also think the new code block style is not as good:

### OLD

Screen Shot 2022-06-26 at 12 17 36

### NEW

Screen Shot 2022-06-26 at 12 17 45

---

It's possible we have some customizations that would result in a better look if they were removed. But we might need to add CSS to tweak things.

**Comment by tcmetzger** [comment]
This is great, thank you!

Some first impressions: 

We might want to consider restoring the CSS color assigned to **headlines and a:hover / a:active**, I don't think the light blue color works with our overall color scheme:

Old version: headlines/links are black (#212529 on white), contrast ratio 15.42:1
image

New version: headlines/links are blue (#459db9 on white), , contrast ratio 3.09:1
image

Similarly, the **inline code** color of the theme (pink) doesn't seem to work well with our color scheme (red for links, for example):

Old version: inline is black on gray (#212529 on #eef0f1), contrast ratio 13.49:1
image

New version: inline is pink (#e83e8c on white), contrast ratio 3.81:1
image

Also, from an accessibility standpoint, the theme's standard colors are not optimal (contrast ratio 3.09:1 for headlines and 3.81:1 for inline code), as WCAG 2.0 level AA requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text. The customized colors we have been using in the past seem to work much better (contrast ratio 15.42:1 for headlines and 13.49:1 for inline code).

**Comment by bryevdv** [comment]
@tcmetzger thanks for that analysis. What tool did you use?

In the interest of maintainability I have stripped down and tried to comment all the sections in `custom.css`. Would you be able to take a quick look at putting back the headers and inline code into the new pared-down `custom.css` ? (And any other tweaks you think might help). Apart from styling I think most things are now acceptably working.

Note I did remove and wanted to ask about the panel header colors. I have to be honest and say after some experience I find them rather distracting, especially when card headers are different heights. I guess I would like to consider something simpler / blander, even if not the current light grey.

File: sphinx/source/_static/custom.css [hunk 1]
```diff
@@ -1,6 +1,17 @@
 /* Customize theme font colors */
-:root {
+html[data-theme=light] {
```
Comment (bryevdv) [inline sphinx/source/_static/custom.css:2]: ahhh that's why it wasn't working....

Comment (tcmetzger) [reply sphinx/source/_static/custom.css:2]: 😊

**Comment by tcmetzger** [comment]
@bryevdv I'd like to look at some more details - I'll set this PR to `ready` when I'm done!

File: sphinx/source/_static/custom.css [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
   --pst-color-borders: 206, 212, 218;
+  --pst-color-primary: var(--pst-color-text-base);
+}
+
+a { color: #CF462A; }
+
+code {
```
[the remark this answers is not in the document]
Comment (bryevdv) [reply sphinx/source/_static/custom.css:9]: Meme we need both? We might also not want to interfere with the highlighted pygment blocks? Not sure

[the remark this answers is not in the document]
Comment (tcmetzger) [reply sphinx/source/_static/custom.css]: I'll make it `.literal .pre` - that seems to work across the different areas of the docs (user guide, ref, etc), but not interfere with the pygment blocks!

**Comment by tcmetzger** [comment]
@bryevdv I made the changes I wanted to make - I don't know why this one integration test is failing but it seems unrelated. Ready to merge, I think!

**Review by tcmetzger — APPROVED** [review]

**Pull request merged**